fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -2777,9 +2777,8 @@ struct StartFullBackupTaskFunc : BackupTaskFuncBase {
 
 		state Reference<TaskFuture> backupFinished = futureBucket->future(tr);
 
-		// Initialize the initial snapshot and create tasks to continually write logs and snapshots
-		// The initial snapshot has a desired duration of 0, meaning go as fast as possible.
-		wait(config.initNewSnapshot(tr, 0));
+		// Initialize the initial snapshot and create tasks to continually write logs and snapshots.
+		wait(config.initNewSnapshot(tr, config.initialSnapshotIntervalSeconds().get(tr).get().orDefault(0)));
```

Comment:
You can't do this.  `config.initialSnapshotIntervalSeconds().get(tr)` returns a Future and you must wait on it to use the result.

Reply:
You could add
`state Future<Optional<int64_t>> initialSnapshotIntervalSeconds = config.initialSnapshotIntervalSeconds().get(tr);`
above
and then add `success(initialSnapshotIntervalSeconds)` to the `wait()` call above that is waiting for all the database reads to finish.  Then, you can use
`initialSnapshotIntervalSeconds.orDefault(-1)`

Note that I used -1 for the default, not 0, because 0 would break the logic in `initNewSnapshot()` that will load the default for this BackupConfig if no value >=0 is passed in.

Reply:
Thanks. One question though: if we use -1 as the default value, `initNewSnapshot ()` will use `snapshotIntervalSeconds`, which is 10 days instead of ASAP. Isn't that changing to the current behavior?

Reply:
Good catch! You are correct, this default should be 0 for this call to `initNewSnapshot()`.  I don't know what I was thinking.